lua/nvim-treesitter/textobjects/move.lua

```diff
@@ -1,21 +1,55 @@
 local ts_utils = require'nvim-treesitter.ts_utils'
-local shared = require'nvim-treesitter.textobjects.shared'
 local attach = require'nvim-treesitter.textobjects.attach'
+local queries = require'nvim-treesitter.query'
 
 local M = {}
 
-function M.goto_adjacent(query_string, forward, start, same_parent, overlapping_range_ok)
-  local bufnr, _, node = shared.textobject_at_point(query_string)
-  local adjacent = shared.get_adjacent(forward, node,  query_string, same_parent, overlapping_range_ok, bufnr)
+local function move(query_string, forward, start, bufnr)
+  local bufnr = bufnr or vim.api.nvim_get_current_buf()
+  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+  row = row - 1
 
-  ts_utils.goto_node(adjacent, not start)
+  local function filter_function(match)
+    local range = {match.node:range()}
```

Comment:
I think we could unpack this to four variables to make it easy to read

Reply:
It really does not make the code better. I've changed to this way since we really had a lot of bugs when one of the unpacked objects had a wrong name. I first had the idea to use LSP ranges to have some structural data but having just one variable instead of four makes live a lot easier and add less visual noise. Even if you can only index the range by numbers and not by start/end
```lua
    local object_row, object_col, end_row, end_col = match.node:range()
    if not start then
      if end_col == 0 then
        object_row = end_row - 1
        object_col = end_col
      else
        object_row = end_row
        object_col = end_col - 1
      end
    end
    if forward then
      return object_row > row or (object_row == row and object_col > col)
    else
      return object_row < row or (object_row == row and object_col < col)
    end
```

Reply:
@stsewd merge or change?

Reply:
yeah, having a structure instead of a simple table should be the correct solution. We can improve that later.